lib/PoolConnection.js

```diff
@@ -29,11 +29,19 @@ PoolConnection.prototype.changeUser = function changeUser(options, callback) {
 
 PoolConnection.prototype.release = function release() {
   var pool = this._pool;
+  var connection = this;
 
   if (!pool || pool._closed) {
     return;
   }
 
+  if (this._purge) {
```

Comment:
This should be in the pool code, not here; `conn.release()` is just a convenience function, and there are users who actually call `pool.releaseConnection(conn)` instead and they'll end up with different behavior.